google/cloud/storage/internal/object_requests.cc

```diff
@@ -421,12 +421,15 @@ std::string UploadChunkRequest::RangeHeader() const {
 UploadChunkRequest UploadChunkRequest::RemainingChunk(
     std::uint64_t new_offset) const {
   UploadChunkRequest result = *this;
-  if (new_offset < offset_) {
+  if (new_offset < offset_ || new_offset >= offset_ + payload_size()) {
     result.offset_ = new_offset;
     result.payload_.clear();
     return result;
   }
-  PopFrontBytes(result.payload_, new_offset - result.offset_);
+  // No chunk can be larger than what fits in memory, and because `new_offset`
+  // is the [offset_, offset_ + payload_size()], this static_cast<> is safe:
```

Comment:
Nit: That interval is open on the end, so `)` would be a better closer.

s/is the/is in/

Reply:
Oops ... auto merge.